portal/views.py

```diff
@@ -301,38 +298,106 @@ def submit_transfer():
 
     transfer = TransferClient(authorizer=authorizer)
 
-    source_endpoint_id = app.config['DATASET_ENDPOINT_ID']
-    source_endpoint_base = app.config['DATASET_ENDPOINT_BASE']
-    destination_endpoint_id = browse_endpoint_form['endpoint_id']
-    destination_folder = browse_endpoint_form.get('folder[0]')
-
     transfer_data = TransferData(transfer_client=transfer,
-                                 source_endpoint=source_endpoint_id,
-                                 destination_endpoint=destination_endpoint_id,
-                                 label=browse_endpoint_form.get('label'))
+                                 source_endpoint=params['source_endpoint_id'],
+                                 destination_endpoint=params['destination_endpoint_id'],
+                                 label=params['label'])
 
     for ds in filtered_datasets:
-        source_path = source_endpoint_base + ds['path']
-        dest_path = browse_endpoint_form['path']
+        source_path = params['source_endpoint_base'] + ds['path']
+        dest_path = params['destination_path']
 
-        if destination_folder:
-            dest_path += destination_folder + '/'
+        if params['destination_path']:
+            dest_path += params['destination_path'] + '/'
 
         dest_path += ds['name'] + '/'
 
         transfer_data.add_item(source_path=source_path,
                                destination_path=dest_path,
                                recursive=True)
 
-    transfer.endpoint_autoactivate(source_endpoint_id)
-    transfer.endpoint_autoactivate(destination_endpoint_id)
+    transfer.endpoint_autoactivate(params['source_endpoint_id'])
+    transfer.endpoint_autoactivate(params['destination_endpoint_id'])
     task_id = transfer.submit_transfer(transfer_data)['task_id']
 
     flash('Transfer request submitted successfully. Task ID: ' + task_id)
 
     return(redirect(url_for('transfer_status', task_id=task_id)))
 
 
+@app.route('/submit-transfer', methods=['GET'])
+@authenticated
+def retry_transfer():
+    transfer_data = session['_inflight_transfer']
+    return transfer_datasets(transfer_data)
```

Comment:
Is this actually retrying or is this to handle the redirect back from auth after doing the consent flow?

Reply:
Probably less of a retry and more of a first attempt after realizing we need an authorization hop... maybe `process_inflight_transfer()` is more explicit?

Reply:
Yeah I think that name is a better description of the function